sdk/tracing/azure-core-tracing-opencensus/README.md

```diff
@@ -0,0 +1,135 @@
+# Azure Tracing client library for Java
+This package enables distributed tracing across Azure SDK Java libraries through [Opencensus][opencensus]. OpenCensus is an open source, vendor-agnostic, single distribution of libraries to provide metrics collection and distributed tracing for services. 
+The Azure core tracing package provides:
+- Context propogation, in order to correlate activities and requests between services with an initial customer action.
+- Tracing user requests to the system, allowing to pinpoint failures and performance issues.
+
+[Source code][source_code] | [API reference documentation][api_documentation] | [Product
+documentation][aztracing_docs] | [Samples][samples]
+
+## Getting started
+### Prerequisites
+- Java Development Kit (JDK) with version 8 or above
+- [Maven][maven]
+```xml
+<dependency>
+    <groupId>io.opencensus</groupId>
+    <artifactId>opencensus-api</artifactId>
+    <version>0.21.0</version>
+</dependency>
+
+<dependency>
+    <groupId>io.opencensus</groupId>
+    <artifactId>opencensus-impl</artifactId>
+    <version>0.21.0</version>
+</dependency>
+
+<dependency>
+    <groupId>io.opencensus</groupId>
+    <artifactId>opencensus-exporter-trace-zipkin</artifactId>
+    <version>0.21.0</version>
+</dependency>
+```
+
+### Adding the package to your product
+```xml
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-core-tracing-opencensus</artifactId>
+  <version>1.0.0-preview.3</version>
+</dependency>
+```
+## Key concepts
+### Trace
+A trace is a tree of spans showing the path of work through a system. A trace on its own is distinguishable by a unique 16 byte sequence called a TraceID.
+### Span
+A span represents a single operation in a trace. A span could be representative of an HTTP request, a remote procedure call (RPC), a database query, or even the path that a code takes.
+
+## Examples
+The following sections provides examples of using the azure-core-tracing-opencensus plugin with some of the Azure Java SDK libraries:
+### Using the plugin package with Http Client libraries
+- Sync create a Secret using [azure-keyvault-secrets][azure-keyvault-secrets] with tracing enabled.
+    
+	Users can additionally pass the value of the current tracing span to the sdk's using key **"opencensus-span"** on the [Context][context] parameter of the calling method:
+	```java
+	try (Scope scope = tracer.spanBuilder("tracing-user-parent").startScopedSpan()) { 
+		secretClient.setSecretWithResponse(new Secret("BankAccountPassword", "f4G34fMh8v"),
+	    		new Context("opencensus-span", tracer.getCurrentSpan()));
+	}
+	```
+- Async Create a Secret using [azure-keyvault-secrets][azure-keyvault-secrets] with tracing enabled
+    
+    Users can additionally pass the value of the current tracing span to the sdk's using key **"opencensus-span"** on the subscriberContext of the calling method:
+	```java
```

Comment:
The use of block() doesn't showcase how to use this in an async way with reactor. I would make use of reactor methods to close the span when it has completed. Same with the snippets below.

Reply:
updated

Reply:
I think you've misunderstood. In a `try`, block it'll close the Span after line 141 is complete. Since the call is asynchronous, that means that after we link up all the triggers in 140 and invoke the call to Key Vault, it'll move onto the next line without waiting to see if the call to key vault completes. And then closes the span prematurely.

We should not be using the `try` block at all for asynchronous calls.